File: qqbot-plugin-host/src/main/java/com/mieai/qqbot/plugin/host/LoadedPluginMetadata.java
package com.mieai.qqbot.plugin.host;

import java.nio.file.Path;
import java.util.Objects;
import java.util.Set;

public record LoadedPluginMetadata(
        String id,
        String name,
        String version,
        String apiCompatibility,
        Path path,
        String sha256,
        String entrypoint,
        String configurationSchema,
        String defaultConfigurationPath,
        String defaultConfiguration,
        Set<String> capabilities) {
    public LoadedPluginMetadata {
        if (defaultConfigurationPath == null || defaultConfigurationPath.isBlank()) {
            throw new IllegalArgumentException("defaultConfigurationPath must not be blank");
        }
        if (defaultConfiguration == null || defaultConfiguration.isBlank()) {
            throw new IllegalArgumentException("defaultConfiguration must not be blank");
        }
        Objects.requireNonNull(path, "path must not be null");
        capabilities = Set.copyOf(capabilities);
    }
}

File: qqbot-plugin-host/src/main/kotlin/com/mieai/qqbot/plugin/host/LoadedPluginMetadata.kt
package com.mieai.qqbot.plugin.host

import java.nio.file.Path
class LoadedPluginMetadata(
    val id: String,
    val name: String,
    val version: String,
    val apiCompatibility: String,
    val path: Path,
    val sha256: String,
    val entrypoint: String,
    val configurationSchema: String,
    val defaultConfigurationPath: String,
    val defaultConfiguration: String,
    capabilities: Set<String>,
) {
    val capabilities: Set<String> = capabilities.toSet()

    init {
        require(defaultConfigurationPath.isNotBlank()) { "defaultConfigurationPath must not be blank" }
        require(defaultConfiguration.isNotBlank()) { "defaultConfiguration must not be blank" }
    }
}
